include/seqan3/range/decorator/gap_decorator.hpp

```diff
@@ -574,8 +727,35 @@ class gap_decorator
         return const_iterator{*this};
     }
 
-    /*!\brief Returns an iterator to the element following the last element of the decorator.
-     * \returns Iterator to the behind last element.
+    /*!\brief Returns a pseudo random access iterator to the first element of the container.
+     * \returns Iterator to the first element.
+     *
+     * If the container is empty, the returned iterator will be equal to end().
+     *
+     * \attention The returned iterator models the std::ranges::RandomAccessIterator but
+     *            **the operations on it do not guarantee constant time**.
+     *
+     * ### Complexity
+     *
+     * Constant.
+     *
+     * ### Exceptions
+     *
+     * No-throw guarantee.
+     */
+    iterator_ra begin_ra() noexcept
```

Comment:
but I guess it looks weird in the docu if there is a const and you return the non-const iterator. So maybe just add a separate const version that returns a const iterator. That would less irritate the people I guess.

Reply:
Ok so the gap decorator iterator is always const, because I do not allow to change the data structure via the index. thats why the gap_decorator did not have a non-const begin in the first place. I can explicitely delete it? and the const version should then return the const iterator of course.

Reply:
so what we need is a `begin_ra() const` and a `cbegin_ra()`, right? Becase it is always a const iterator. But the begin_ra is still not marked const yet?

Reply:
yes sorry I wasn't careful with the changes.
we are missing the const.
and we can discuss if we should remove the `const_iterator`